Mathlib/Condensed/TopComparison.lean

```diff
@@ -0,0 +1,170 @@
+/-
+Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.CategoryTheory.Limits.Preserves.Opposites
+import Mathlib.Condensed.Explicit
+
+/-!
+
+# The functor from topological spaces to condensed sets
+
+This file develops some API for "topologically concrete" categories, defining universe polymorphic
+"Yoneda presheaves" on such categories. If the forgetful functor to `TopCat` has nice properties,
+like preserving pullbacks and finite coproducts, then this Yoneda presheaf satisfies the sheaf
+condition for the regular and extensive topologies respectively.
+
+We then apply this API to `CompHaus` and define the functor
+`topCatToCondensed : TopCat.{u+1} ⥤ CondensedSet.{u}`.
+
+## Projects
+
+* Prove that `topCatToCondensed` is faithful.
+* Define compactly generated topological spaces.
+* Prove that `topCatToCondensed` restricted to compactly generated spaces is fully faithful.
+* Define the left adjoint of the restriction mentioned in the previous point.
+-/
+
+universe w w' v u
+
+open CategoryTheory Opposite Limits regularCoverage
+
+variable {C : Type u} [Category.{v} C] (F : C ⥤ TopCat.{w}) (Y : Type w') [TopologicalSpace Y]
+-- `C` is meant to be `CompHaus`, `Profinite` or `Stonean`.
+
+namespace ContinuousMap
+
+/--
+A universe polymorphic "Yoneda presheaf" on `C` given by continuous maps into a topoological space
+`Y`.
+-/
+@[simps]
+def yonedaPresheaf : Cᵒᵖ ⥤ Type (max w w') where
+  obj X := C(F.obj (unop X), Y)
+  map f g := ContinuousMap.comp g (F.map f.unop)
+
+/--
+A universe polymorphic Yoneda presheaf on `TopCat` given by continuous maps into a topoological
+space `Y`.
+-/
+@[simps]
+def yonedaPresheaf' : TopCat.{w}ᵒᵖ ⥤ Type (max w w') where
```

Comment:
I'm slightly afraid that at some point in the future someone will look for something like this in a file called something like `Mathlib/Topology/Category/TopCat/Yoneda.lean`, not find anything and then rebuild it, so that we end up with multiple developments of this concept. Do you think that it would make sense to move the parts of this file that are not specific to condensed sets into `Topology/Category/TopCat/*`?